src/solvers/umfpack.jl

```diff
@@ -230,13 +256,34 @@ UmfpackWS(F::ATLU, refinement::Bool=has_refinement(F)) = UmfpackWS(F.parent, ref
     
 """
     copy(F::UmfpackLU, [ws::UmfpackWS]) -> UmfpackLU
-A shallow copy of UmfpackLU to use in multithreaded applications. This function duplicates the working space, control and locks.
-It can also take transposed or adjoint `UmfpackLU`s.
+A shallow copy of UmfpackLU to use in multithreaded solve applications. 
+This function duplicates the working space, control, info and lock fields.
+
+Warning: This shallow copy should not be used for parallel factorizations or re-factorizations,
+doing so may lead to race-conditions.
 """
 # Not using simlar helps if the actual needed size has changed as it would need to be resized again
-Base.copy(F::UmfpackLU, ws=UmfpackWS(F)) = UmfpackLU(
-    F.symbolic,
-    F.numeric,
+Base.copy(F::UmfpackLU, ws=UmfpackWS(F)) = 
+    UmfpackLU(
+        F.symbolic,
+        F.numeric,
+        F.m, F.n,
+        F.colptr,
+        F.rowval,
+        F.nzval,
+        F.status,
+        ws,
+        copy(F.control),
+        copy(F.info),
+        ReentrantLock()
+    )
+Base.copy(F::T, ws=UmfpackWS(F)) where {T <: ATLU} = 
+    T(copy(parent(F), ws))
+
+Base.deepcopy(F::UmfpackLU{Tv, Ti}, ws=UmfpackWS(F)) where {Tv, Ti} = 
```

Comment:
There are two problems with this. First, see the `deepcopy` doc string:
```
While it isn't normally necessary, user-defined types can override the default `deepcopy`
behavior by defining a specialized version of the function
`deepcopy_internal(x::T, dict::IdDict)`
```

As it is, this method won't be called for UmfpackLUs inside other objects.

Second, `deepcopy` does not generically accept a second argument. To add the `ws` argument you need to create a new function, e.g. `deepcopy_umfpacklu` to accept it.

Reply:
Okay I'll get those fixed, thanks!